### PR TITLE
Refactoring of documentation to rST and reorg autodocs

### PR DESCRIPTION
**Pull request by forbearer**
Here's what the updated rtfd result looks: https://petastorm.readthedocs.io/en/docs_refactor/

Key changes of note:
* Sphinx auto doc support files got moved into `docs/autodoc/`
* README and 3 markdown files under `docs/` are converted to rST files. Let's make sure we'll be happy with this going forward...or, we can opt to rollback on this attempt.

Oh, I just realized there is at least one README.md file missed in the repo hierarchy!

### CODE REVIEW

**Comment by forbearer** [comment]
I redid my commits so that the 4 file renames are done first. This way, a reviewer can look at just the MD->RST commit to look at changes for the 4 .md files (e.g., https://github.com/uber/petastorm/pull/123/commits/2513a99ab687808c383678b03b367b564ec6f48a ).

File: docs/autodoc/reader.rst [hunk 1]
```diff
@@ -1,51 +1,51 @@
-The Reader API
-==============
+``reader``
```
Comment (selitvin) [inline docs/autodoc/reader.rst]: This now looks like code ("`"). As a reader, I would expect to find corresponding code elements (packages or classes). Should't it be the same style as before?

Comment (forbearer) [reply docs/autodoc/reader.rst]: Oh yeah, for side by side comparison:
* Master: https://petastorm.readthedocs.io/en/latest/
* New look: https://petastorm.readthedocs.io/en/docs_refactor/

Comment (selitvin) [reply docs/autodoc/reader.rst]: In the example you refer to these are the package names, hence it makes sense to use this style. In our case, it's just 'headings'. We should improve our package structure so they correspond, I agree. But for now, IMO it's a little confusing.

Comment (forbearer) [reply docs/autodoc/reader.rst]: Ah, that's a valid point.  I may have gone too far in the name of consistency.

Shall I restore the original non-code-block headings?

Comment (forbearer) [reply docs/autodoc/reader.rst]: After Slack convo, we concluded that the original heading style is better for now.

So I've restored the heading style and pushed that.

**Pull request merged**